File: 003/003.py
class Solution:
    def lengthOfLongestSubstring(self, s):

        ## Input:
        # s: str
        ## Output:
        # int

        repeat = {}
        count = 0
        max_count = 0

        i, j = 0, 0
        while (j < len(s)):
            if (s[j] not in repeat):
                count += 1
                repeat[s[j]] = j
                j += 1
            else:
                index = repeat[s[j]]
                while (i <= index):
                    del repeat[s[i]]
                    i += 1
                repeat[s[j]] = j
                j += 1
                if (count > max_count):
                    max_count = count
                count = j - i 

        if (count > max_count):
            max_count = count               
        
        return max_count

File: 003/003.cpp
class Solution
{
public:
    int lengthOfLongestSubstring(string s)
    {
        map<char, int> repeat;
        int count = 0;
        int max_count = 0;

        int i = 0;
        int j = 0;
        while (j < s.length())
        {
            if (repeat.find(s[j]) == repeat.end())
            {
                count++;
                repeat[s[j]] = j;
                j++;
            }
            else
            {
                int index = repeat[s[j]];
                while (i <= index)
                {
                    repeat.erase(s[i]);
                    i++;
                }
                repeat[s[j]] = j;
                j++;
                if (count > max_count)
                {
                    max_count = count;
                }
                count = j - i;
            }
        }

        if (count > max_count)
        {
            max_count = count;
        }

        return max_count;
    }
};